shared/src/main/scala/org/atnos/eff/Arrs.scala

```diff
@@ -69,20 +73,32 @@ case class Arrs[R, A, B](functions: Vector[Any => Eff[R, Any]]) extends (A => Ef
   }
 
   def contramap[C](f: C => A): Arrs[R, C, B] =
-    Arrs(((c: Any) => Eff.pure[R, Any](f(c.asInstanceOf[C]).asInstanceOf[Any])) +: functions)
+    Arrs(((c: Any) => Eff.pure[R, Any](f(c.asInstanceOf[C]).asInstanceOf[Any])) +: functions, onNone)
 
   def contraFlatMap[C](f: C => Eff[R, A]): Arrs[R, C, B] =
-    Arrs(f.asInstanceOf[Any => Eff[R, Any]] +: functions)
+    Arrs(f.asInstanceOf[Any => Eff[R, Any]] +: functions, onNone)
+
+  /** adapt the input and output of an Arrs function */
+  def dimap[C, D](f: C => A)(g: Eff[R, B] => Eff[R, D]): Arrs[R, C, D] =
```

Comment:
Sorry, but this ain't `dimap` ;) `g` has to be `B => D` for it to be `dimap`. No reason not to include this though: maybe `dimapEff`?